### ISSUE TITLE
Different behaviour of undesirable linters with empty character vectors

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1867]
I'd have expected both of these arguments to have similar behaviour. Is this expected?

``` r
library(lintr)

invisible(undesirable_function_linter(fun = character(0L)))

undesirable_operator_linter(op = character(0L))
#> Error in undesirable_operator_linter(op = character(0L)): 'op' should be a named character vector; use missing elements to indicate default messages.
```

<sup>Created on 2022-12-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by AshesITR** [comment]
Agree. Both of them should error because there should be no need to create a no-op linter.
